packages/eas-cli/src/commands/channel/create.ts

```diff
@@ -0,0 +1,127 @@
+import { getConfig } from '@expo/config';
+import { Command, flags } from '@oclif/command';
+import chalk from 'chalk';
+import gql from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../../graphql/client';
+import log from '../../log';
+import { ensureProjectExistsAsync } from '../../project/ensureProjectExists';
+import { findProjectRootAsync, getProjectAccountNameAsync } from '../../project/projectUtils';
+import { promptAsync } from '../../prompts';
+
+type UpdateChannel = {
+  id: string;
+  channelName: string;
+};
+
+async function createUpdateChannelOnAppAsync({
+  appId,
+  channelName,
+}: {
+  appId: string;
+  channelName: string;
+}): Promise<UpdateChannel> {
+  const releaseMapping = JSON.stringify({ data: [], version: 0 });
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<
+        { updateChannel: { createUpdateChannelForApp: UpdateChannel } },
+        { appId: string; channelName: string; releaseMapping: string }
+      >(
+        gql`
+          mutation CreateUpdateChannelForApp(
+            $appId: ID!
+            $channelName: String!
+            $releaseMapping: String!
+          ) {
+            updateChannel {
+              createUpdateChannelForApp(
+                appId: $appId
+                channelName: $channelName
+                releaseMapping: $releaseMapping
+              ) {
+                id
+                channelName
+                releaseMapping
+              }
+            }
+          }
+        `,
+        {
+          appId,
+          channelName,
+          releaseMapping,
+        }
+      )
+      .toPromise()
+  );
+  return data.updateChannel.createUpdateChannelForApp;
+}
+
+export default class Channel extends Command {
+  static description = 'Create a channel on the current project.';
+
+  static args = [
+    {
+      name: 'channelName',
+      required: false,
+      description: 'Name of the channel to create',
+    },
+  ];
+
+  static flags = {
+    json: flags.boolean({
+      description: 'return a json with the new channel ID, name and release mapping.',
+      default: false,
+    }),
+  };
+
+  async run() {
+    let {
+      args: { channelName },
+      flags: { json: jsonFlag },
+    } = this.parse(Channel);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const accountName = await getProjectAccountNameAsync(projectDir);
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!channelName) {
+      const validationMessage = 'Channel name may not be empty.';
+      if (jsonFlag) {
```

Comment:
```suggestion
      if (flags.json) {
```
No need to assign this to a separate variable, I think.